types/jest/index.d.ts

```diff
@@ -973,7 +922,7 @@ declare namespace jest {
          * expect(desiredHouse).toMatchObject<House>(...standardHouse, kitchen: {area: 20}) // wherein standardHouse is some base object of type House
          */
         // tslint:disable-next-line: no-unnecessary-generics
-        toMatchObject<E extends {} | any[]>(expected: E): R;
+        toMatchObject<E extends T | any[]>(expected: E): R;
```

Comment:
This is incorrect.
This change will have a typescript error for the example test for toMatchObject
https://jestjs.io/docs/en/expect#tomatchobjectobject

Reply:
I guess it needs to be Partial<T>, I can revert this one if it can unblock the PR from getting merged

Reply:
As it is I expect that it will prevent the pull request from being merged.  As to what the correct typing should be you will have to check the source.
https://github.com/facebook/jest/blob/baf9f9937720e87d2b2bd09f4c053fa4f16424ec/packages/expect/src/matchers.ts#L922
`const pass = equals(received, expected, [iterableEquality, subsetEquality]);`
https://github.com/facebook/jest/blob/baf9f9937720e87d2b2bd09f4c053fa4f16424ec/packages/expect/src/jasmineUtils.ts#L37
```
export function equals(
  a: unknown,
  b: unknown,
  customTesters?: Array<Tester>,
  strictCheck?: boolean,
): boolean {
  customTesters = customTesters || [];
  return eq(a, b, [], [], customTesters, strictCheck ? hasKey : hasDefinedKey);
}
```
https://github.com/facebook/jest/blob/baf9f9937720e87d2b2bd09f4c053fa4f16424ec/packages/expect/src/jasmineUtils.ts#L65

Reply:
I've reverted this change